airbyte-integrations/connectors/destination-dev-null/build.gradle

```diff
@@ -17,7 +17,18 @@ application {
     //applicationDefaultJvmArgs = ['-XX:+ExitOnOutOfMemoryError', '-XX:MaxRAMPercentage=75.0', '--add-opens', 'java.base/java.lang=ALL-UNNAMED']
 }
 
+dependencies {
+    // temporary dependency so that we can continue running the legacy test suite.
+    // eventually we should remove this + rely solely on the bulk CDK tests.
+    integrationTestJavaImplementation testFixtures(project(":airbyte-cdk:java:airbyte-cdk:airbyte-cdk-db-destinations"))
+    // integrationTestJavaImplementation testFixtures("io.airbyte.cdk:airbyte-cdk-db-destinations:0.47.0")
+}
+
 // Uncomment to run locally
 //run {
 //    standardInput = System.in
 //}
+
+integrationTestJava {
+    dependsOn buildConnectorImage
```

Comment:
this probably isn't what we want long term, but it works for now :shrug:

(.... I'm a bit confused as to why I needed to add this explicitly, but at least locally, I wasn't getting a docker image without it)

Reply:
You needed to add it because the `integrationTestJava` task I added to the airbyte-bulk-connector plugin does not depend on `buildConnectorImage` (specifically, because it does not depend on `assembly`).

What we really want is

```
integrationTest (dummy task) -> integrationTestKotlin -> check
                            \-> integrationTestDocker -> check, assembly
```

AND `airbyte-ci` runs `integrationTest` (maybe if `integrationTestJava` isn't available, so we don't break stuff).

And then I guess

`src/test-integration` => integrationTestKotlin
`src/test-integration-docker` => integrationTestDocker

?

Reply:
well, this is currently failing b/c `buildConnectorImage` doesn't even exist on CI >.> so I guess I need to figure out how airbyte-ci is actually doing stuff under the hood now.

I do generally agree that separating the new+old tests makes sense, will probably punt on that particular yak for at least a few days though

Reply:
Weird! If I run `:build` I can see the connector image getting built AND specifically I can see `:buildConnectorImage` get run.
![Screenshot 2024-10-01 at 9 38 45 AM](https://github.com/user-attachments/assets/dc27cde9-23b2-4b2b-bedb-ea7af9481281)

Also
```
jschmidt@Johnny-Schmidt-MacBook-Pro---C2WMR2G02N airbyte % ./gradlew :airbyte-integrations:connectors:destination-dev-null:buildConnectorImage --dry-run
:airbyte-integrations:connectors:checkPython SKIPPED
:airbyte-integrations:connectors:pipInstall SKIPPED
:airbyte-integrations:connectors:poetryInstallAirbyteCI SKIPPED
:airbyte-integrations:connectors:destination-dev-null:checkKotlinGradlePluginConfigurationErrors SKIPPED
...
```

Or is there a subtlety here that I'm missing?

Regardless, I think the right thing to do is make it depend on `assemble`. That's what the old plugin did.